docs/how-to/uc-image-creation.rst

```diff
@@ -0,0 +1,214 @@
+How to create a real-time Ubuntu Core image
+===========================================
+
+.. tip::
+
+    This guide assumes you have access to features available exclusively to `IoT App Store`_ users.
+
+The :doc:`../how-to/uc-boot-parameters` describes the method for dynamically configuring the system for real-time processing.
+The configuration is an iterative process that is best done together with the expected workload. 
+Once satisfied with the configurations, it is time to prepare the operating system for going into production.
+
+This guide shows how to statically set the desired Kernel command-line options for the Ubuntu Core system.
+
+To do this, we need to create a custom gadget snap, create a model assertion, and then build the OS image.
+
+
+Create the gadget snap
+----------------------
+
+The `gadget snap`_ documentation is a recommended read before starting.
+
+This is best done by forking an existing reference gadget, then changing it for our purpose.
+For example, there is the `pc gadget`_ which is suitable for most AMD64 platforms, and the `pi gadget`_ which is meant for Raspberry Pis.
+
+Let's create a custom gadget snap based on the core22 pc gadget.
+
+Clone the specific branch and enter the directory:
+
+.. code-block:: shell
+
+    git clone https://github.com/snapcore/pc-gadget.git --branch=22 --depth=1
+    cd pc-gadget
+
+
+Add the desired kernel command line in an array to ``kernel-cmdline.append`` in ``gadget/gadget-amd64.yaml``.
+For example:
+
+.. code-block:: yaml
+
+    kernel-cmdline:
+        append:
+            - nohz=on
+            - nohz_full=2-N
+            - irqaffinity=0-1
+
+
+Refer to :doc:`../reference/kernel-boot-parameters` for the list of supported parameters.
+
+Modify ``snapcraft.yaml`` to fit your application.
+At least, make sure to change the name and version to something distinct, for example to ``realtime-pc`` and ``example`` respectively.
+
+
+Now, build the gadget snap:
+
+.. code-block:: console
+
+    $ snapcraft --verbose
+    ...
+    Created snap package realtime-pc_example_amd64.snap
+
+
+.. tip::
+    You need to rebuild the snap every time you change the `snapcraft.yaml` file.
+
+
+Create the model assertion
+--------------------------
+
+The model assertion is a digitally signed document that describes the content of the Ubuntu Core image.
+Read the `model assertion`_ documentation before continuing.
+
+Below is an example model assertion in YAML, describing a `core22` Ubuntu Core
+image:
+
+.. literalinclude :: uc-image-creation/model.json
+   :language: json
+
+Create the model assertion in the same directory as the pc-gadget.
+
+Set the following:
+
+
+- ``authority-id``, ``brand-id`` to your developer ID
+- ``timestamp`` to a RFC3339 formatted string, withing you signing key's validity
+- ``store`` to you IoT App Store ID
+
+Change the value of ``model`` to something representative of your model.
```

Comment:
I think we want to also mention that they may want to add a new model in their Serial Vault as well if they want the devices they make using this document to be able to connect to their Store (i.e. test updates to the realtime-kernel snap).

Reply:
Regardless of adding the model to serial vault, aren't the devices going to receive updates for all the snaps that are in the stores?

Reply:
In order for a device to see any store, it needs a serial assertion. A generic one if they go to the Global Store, or one from the Partner Serial Vault if they use their Brand. In that latter case, the devices will see their Store and receive those updates.

But a device can only get a serial assertion from the PSV just in case some key information checks out; the model is signed by a key owned by the Brand account, the device provides a serial number in its request to the PSV for a serial assertion, **and** the model name and secret the gadget uses match the model name and secret in the PSV. Customers will know these details because we'll have given them an onboarding guide, that guide also mentions [this](https://ubuntu.com/core/services/guide/serial-vault-overview) page which you can call out to.

Reply:
Thanks for explaining. Very helpful. I've added minimal a note: https://github.com/canonical/real-time-ubuntu-docs/pull/51/commits/24255ec6f79a1c42a73a76eb36e8d7ed86840c8b

I was just informed that docs under "core/services" are to be refactored in favor of those under [core/docs](https://ubuntu.com/core/docs). 

I prefer to avoid an in-depth explanation of how everything works inside the real-time doc and instead link to existing docs from Ubuntu Core or Snap Store (whenerver that I find something suitable).